libs/utils/env.py

```diff
@@ -474,19 +485,36 @@ def _init_target(self, force = False):
             # Brackets are there to let the output dir be created automatically
                                     '--workload-automation-vio={}'])
 
+            # Add variant suffix
+            suffix = os.path.splitext(os.path.basename(
+                     system_info['platform']['description']))[0]
+            board_file = self.conf['board'].lower() + suffix + '.json';
+            self._board_path = os.path.join(basepath, 'libs/utils/platforms',
+                                      board_file)
+            if not os.path.exists(self._board_path):
+                self._log.warning('Board descriptor file %s was not found.',
+                                  self._board_path)
+                self._board_path = None
+                board = None
+            else:
+                board = self._load_board()
```

Comment:
The `board_file` is required to both load board information and to obtain the `nrg_model`. Previously, both `_load_board` and `_load_em` would obtain the information they needed by constructing the file path from the `board` field of the target configuration. For GEM5 that's not enough and requires part of the platform description as suffix to identify the proper file. So if I only make the change you suggested, I'll have to construct this path (based on a suffix which is convoluted and prone to error - `self.conf['gem5_settings]['system']['platfomr']['description'])` multiple times:
- when `_load_board(board_file)` is called
- when `_load_em(board_file)` is called
- (in future commits) when self.nrg_model is initialised with is based on an energy file (similar to what we have now: juno_energy.py) that will use the same suffix.

In my opinion is more clear to establish this `_board_path` once.
In the current patch (for each of your points 1., 2., above):
- `_load_board()` will use `_board_path` if is set and if the file exists. It will do nothing otherwise (so we maintain the same behaviour as before)
- it will set up a default `_board_path` which is `<board_name>.json` as it was before, and it will overwrite it for GEM5 that needs to add a suffix.

Yes, this dance is not mandatory (I can obtain that suffix multiple times, although it's ugly) but I thought this solution was cleaner for both the existing code and the one added for GEM5.